### PR TITLE
Rework plugin stopping functionality

### PR DESCRIPTION
**Pull request by disaykin**
- more clean debug logs
 - do not stop already stopped plugins
 - implement common logic in base class

### CODE REVIEW

File: src/plugins/plugins_ex.h [hunk 1]
```diff
@@ -415,7 +415,7 @@ class pluginex : public plugin
 
     pluginex(drakvuf_t drakvuf, output_format_t output);
     virtual ~pluginex();
-    virtual bool stop();
+    virtual bool stop_impl() override;
```
Comment (tklengyel) [inline src/plugins/plugins_ex.h:418]: Why change the name from stop to stop_impl?

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/plugins/plugins_ex.h (hunk 1 above)
Comment (disaykin) [reply src/plugins/plugins_ex.h:418]: I imlement common stopping logic in method `stop` of the base class `plugin` using design pattern `template method`. Derived classes must implement only specific bussiness logic in `stop_impl` pure virtual method.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by disaykin** [comment]
@tklengyel Possibly I fix memory leak in dkommon plugin

**Pull request merged**